### PR TITLE
feat(Core/Conf): Miss Chance Multiplier

### PR DESCRIPTION
**Pull request by Nefertumm**
## Changes Proposed:
-  Add a config to only affect players.

## Issues Addressed:

- Closes 

## Tests Performed:

- Not tested.

## How to Test the Changes:

1. Make 2 characters with notorious level difference
2. Set up the configs
3. Test different cases.

### CODE REVIEW

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by Lynethris** [comment]
I don't if it's impact the PR but 
Screenshot 2022-03-06 181659
:

**Comment by Lynethris** [comment]
I'm not sure what does exactly this PR but the main feature (affect player) doesn't work (maybe cause by the missing option).

 Used a lvl 60 against a lvl 70 in melee and at distance: 

WoWScrnShot_030622_182554

WoWScrnShot_030622_182710

**Comment by Lynethris** [comment]
Tested and not work with a lvl 60 (paladin) against a lvl 70 (melee and spell cast): 

- [ ] Wild PVP
- [ ] Duel
- [ ] BG (testing in progress)

Images sources:
<details>
WoWScrnShot_030622_192134


WoWScrnShot_030622_192147

</details>

File: src/server/worldserver/worldserver.conf.dist [hunk 1]
```diff
@@ -2550,22 +2550,27 @@ Die.Command.Mode = 1
...
 #                     Note: this does not affect when the player is less than 3 levels different than the target,
 #                     where this (linear) formula is used instead to calculate the hit chance: 96 - levelDiff.
+#                     You can set Rate.MissChanceMultiplier.OnlyAffectsPlayer to 1 if you only want to affect the MissChance
+#                     for player casters only. This way you won't be affecting creature missing chance.
 #
 #                     Example: if you want the chance to keep growing linearly, use 1.
 #
 #        Default:     Rate.MissChanceMultiplier.TargetCreature = 11
 #                     Rate.MissChanceMultiplier.TargetPlayer = 7
+#                     Rate.MissChanceMultiplier.OnlyAffectsPvP = 0
```
Comment (Nyeriah) [inline src/server/worldserver/worldserver.conf.dist:2568]:
```suggestion
#                     Rate.MissChanceMultiplier.OnlyAffectsPlayer = 0
```

**Comment by Nyeriah** [comment]
> I don't know if it's impact the PR but : Screenshot 2022-03-06 181659

Did this still show up when you tested it?

**Comment by Lynethris** [comment]
> > I don't know if it's impact the PR but : Screenshot 2022-03-06 181659
> 
> Did this still show up when you tested it?

No

**Comment by Nefertumm** [comment]
Strange then, did the option work without the PR?

**Comment by Lynethris** [comment]
> Strange then, did the option work without the PR?

Without the PR (but keeping the same worldserver with option), it's work. The miss chance affect the player.

**Comment by Lynethris** [comment]
Ok, as I'm not totaly sure to understand how the system work, I posted here all my test step :

I played with a druid lvl 20.

<ins>Without option (Rate.MissChanceMultiplier.OnlyAffectsPlayer = 0):</ins>
- TargetCreature = 40 --> Always miss against a scorpion lvl46 (in blastland)
- TargetPlayer = 40 --> Always miss against a orc lvl30 (2nd account)
- TargetCreature = 2 --> Hit sometime a scorpion lvl46
- TargetPlayer = 2 --> Hit almost time a orc lvl30

<ins>With option (Rate.MissChanceMultiplier.OnlyAffectsPlayer = 1): </ins>
I got the same result that without this option. (And if I understood, whatever the values of targetCreature, it is supposed to be ignored ? If it the case, so Rate.MissChanceMultiplier.OnlyAffectsPlayer doesn't work)

**Comment by temperrr** [comment]
Tested, works

Rate.MissChanceMultiplier.TargetCreature = 11
Rate.MissChanceMultiplier.TargetPlayer = 1
Rate.MissChanceMultiplier.OnlyAffectsPlayer = 1

- [x] Alliance level 40
- [x] Horde level 30
- [x] Horde misses but not that much against level 40 alliance
- [x] horde misses alot vs level 40 creature.

**Pull request merged**